widget_driver/test/drivable_widget_test.dart

```diff
@@ -14,57 +15,165 @@ void main() {
+        when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+
+        final testContainerDrivableWidget = TestContainerDrivableWidget(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+        );
+        await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
 
-      final testContainerDrivableWidget = TestContainerDrivableWidget(
-        environmentInfo: _mockRuntimeEnvironmentInfo,
-      );
-      await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
+        final driver = testContainerDrivableWidget.driver;
+        expect(driver.aTestText, 'A test text');
+      });
+    });
 
-      final driver = testContainerDrivableWidget.driver;
-      expect(driver.aTestText, 'A test text');
+    group('Providable Properties: ', () {
+      testWidgets('Calling `aTestMethod` on driver makes widget reload with new data', (WidgetTester tester) async {
+        when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+
+        final testContainerDrivableWidget = TestContainerDrivableWidget(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+        );
+        await tester.pumpWidget(MaterialApp(home: testContainerDrivableWidget));
+        final driver = testContainerDrivableWidget.driver;
+
+        expect(find.text('didCallTestMethod: false'), findsOneWidget);
+
+        driver.aTestMethod();
+        await tester.pump();
+
+        expect(find.text('didCallTestMethod: true'), findsOneWidget);
+      });
+
+      testWidgets('updateDriverProvidedProperties gets called on widget configuration change',
+          (WidgetTester tester) async {
+        when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+
+        late TestContainerDriver driver;
+        final testContainerDrivableWidget = WrappedTestContainer(
+          environmentInfo: _mockRuntimeEnvironmentInfo,
+          driverCallback: (newDriver) => driver = newDriver,
+        );
+        await tester.pumpWidget(testContainerDrivableWidget);
+        final firstDriver = driver;
+
+        expect(driver.numberOfCallsToUpdateDriverProvidedProperties, 0);
+        expect(driver.someData, 0);
+        expect(find.text('justSomeData: 0'), findsOneWidget);
+
+        await tester.tap(
+          find.byKey(
+            const Key(
+              'wrapped_test_container_button',
+            ),
+          ),
+        );
+        await tester.pumpAndSettle();
+
+        expect(driver.numberOfCallsToUpdateDriverProvidedProperties, 1);
+        expect(driver.someData, 1);
+        expect(find.text('justSomeData: 1'), findsOneWidget);
+        expect(driver, firstDriver);
+      });
     });
 
-    testWidgets('Calling `aTestMethod` on driver makes widget reload with new data', (WidgetTester tester) async {
-      when(() => _mockRuntimeEnvironmentInfo.isRunningInTestEnvironment()).thenReturn(false);
+    group('Driver lifecycle', () {
```

Comment:
Just a small hint.. To these "group" names.. I think it makes for better reading to always end them with a `:`
If you look at the logs output from running your Tests it will print this format:
`{group name} {test name}`

So if you have no separation then you only get that whitespace as separation.
So output is like this:
`Driver lifecycle does not create new driver, if...`
But maybe it is easier to read this test like this:
`Driver lifecycle: Does not create new driver, if...`
Then you can read the second sentance.. and you know it relates to `Driver lifecycle`